src/c++/uds/src/uds/thread-utils.h

```diff
@@ -8,20 +8,19 @@
 
 #include <linux/atomic.h>
 #ifdef __KERNEL__
+#ifdef TEST_INTERNAL
 #include <linux/delay.h>
-#include <linux/jiffies.h>
-#include <linux/mutex.h>
 #include <linux/semaphore.h>
+#endif /* TEST_INTERNAL */
 #else
-#include <linux/mutex.h>
 #include <pthread.h>
 #include <sched.h>
 #include <semaphore.h>
 #include <stdbool.h>
-#endif
+#endif /* __KERNEL__ */
+#if !(defined __KERNEL__) || defined(TEST_INTERNAL)
 
 #include "errors.h"
```

Comment:
The original patch removed the include of errors.h. So does this need to be within an ifdef?

Reply:
It is in an #if clause; I moved the #if line above the errors.h include. This is need fomr the uds_initialize_semaphore() function among others, but I could potentially get away with making it just TEST_INTERNAL.